ducktape/cluster/vagrant.py

```diff
@@ -24,10 +28,18 @@ class VagrantCluster(JsonCluster):
     advertised, this assumes that the Vagrant VM's name is a routeable hostname on all the hosts.
     """
 
-    def __init__(self):
+    def __init__(self, *args, **kwargs):
         hostname, ssh_hostname, username, flags = None, None, None, ""
         nodes = []
 
+        if kwargs.get("use_cluster_info", False):
```

Comment:
@granders It is in `VagrantCluster`, because conceptually this cached file is written a running Ducktape with `VagrantCluster`. It probably makes sense to read/write the file in the same class. 

Another reason is that, if this is in the `JsonCluster`, user needs to specify both `--cluster=ducktape.cluster.json.JsonCluster` and `--use-cluster-info`, which is inconvenient.

Actually I share with you the opinion that this functionality here is very similar to the one in `JsonCluster`. I have provided one solution to consolidate the two, in the reply to @ewencp 's comment. Can you have a look?